src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -319,6 +319,11 @@ def notify_query_listeners(self, query_record: QueryRecord) -> None:
         for listener in self._query_listener:
             listener._add_query(query_record)
 
+    def notify_debug_info_listeners(self, debug_info: Optional[Dict[str, Any]]):
+        if debug_info:
```

Comment:
I prefer the null check to happen outside the function, if `debug_info` is `None`, this function will be a noop.